redirect/main.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  name_prefix = var.name_prefix != null ? var.name_prefix : "${var.project}-${var.environment}"
+
+  default_tags = {
+    Project     = var.project
+    Environment = var.environment
+  }
+
+  tags = merge(local.default_tags, var.tags)
+}
+
+resource "aws_s3_bucket" "redirect" {
+  count = var.create ? 1 : 0
+
+  bucket = var.bucket != null ? var.bucket : "${local.name_prefix}-redirect"
+  acl    = "public-read"
+  tags   = local.tags
+
+  website {
+    redirect_all_requests_to = var.target
+  }
+}
+
+locals {
+  website_endpoint = var.create ? aws_s3_bucket.redirect[0].website_endpoint : ""
+}
+
+resource "aws_cloudfront_distribution" "redirect" {
+  count = var.create ? 1 : 0
+
+  origin {
+    domain_name = local.website_endpoint
+    origin_id   = aws_s3_bucket.redirect[0].id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
+  }
+
+  aliases         = var.domains
+  enabled         = true
+  is_ipv6_enabled = true
+  price_class     = var.cloudfront_price_class
+
+  default_cache_behavior {
+    allowed_methods        = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods         = ["GET", "HEAD"]
```

Comment:
you might want to setup caching TTLs as well since I don't expect S3 to provide a `cache-control` header for the redirect

Reply:
also I'd make sure the redirect doesn't use `301 Moved Permanently` or `308 Permanent Redirect`, since these are hard to change as the name suggests :wink:

Reply:
As discussed, introduced flexibility to change the response code, by default 302.